tag/tag.go

```diff
@@ -100,3 +100,34 @@ func isValidResourceName(name string) bool {
 func forceResourceName(name string) string {
 	return invalidNameRegexp.ReplaceAllString(name, "_")
 }
+
+// GetOtherTags used to check other possible tag attributes on resources
+func GetOtherTags(provider string, srd *schema.ResourceData, filterTag Tag) (string, bool) {
+	// keep the same logic as r.data.GetOk
+	otherTagsMap := make(map[string]string)
+
+	// Special Tag attribute
+	if provider == "aws" {
+		// Some resource like aws_autoscaling_group do not have tags map but tag schema.Set
+		// https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#tag
```

Comment:
I would potentially also add https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#tag-and-tags as it documents the actual format it has which is more important in this case.